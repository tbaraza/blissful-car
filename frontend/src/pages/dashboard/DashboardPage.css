.dashboard-container {
  padding: 2rem;
}

.eye-icon-container {
  display: flex;
  background: #2591e1;
  justify-content: center;
  height: 12rem;
  align-items: center;
}

.user-icon-container {
  display: flex;
  background: #c73a3e;
  justify-content: center;
  height: 12rem;
  align-items: center;
}

.search-icon-container {
  display: flex;
  background: #5fa547;
  justify-content: center;
  height: 12rem;
  align-items: center;
}
