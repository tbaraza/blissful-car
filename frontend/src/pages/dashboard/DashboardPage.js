import React, { Component } from 'react';
import { Card, Icon, List } from 'antd';
import socketIOClient from 'socket.io-client';
import './DashboardPage.css';

const { Meta } = Card;

class DashboardPage extends Component {
  constructor(props) {
    super(props);

    this.state = {
      endpoint: 'http://127.0.0.1:9000',
      pages: {},
      referrers: 0,
      activeUsers: 0,
      searches: 0,
      visits: 0,
    };
  }

  getPageStats = (visitorData) => {
    const { pages, referrers, activeUsers } = visitorData;

    this.setState({
      activeUsers,
      referrers: Object.keys(referrers).length,
      pages,
      visits: Object.keys(pages).length,
    });
  };

  getSearchStats = (searchData) => {
    const { searches } = searchData;
    this.setState({
      searches,
    });
  };

  render() {
    const {
      pages, activeUsers, endpoint, searches, visits,
    } = this.state;
    const socket = socketIOClient(endpoint);
    socket.on('updated-stats', (data) => {
      this.getPageStats(data);
    });

    socket.on('search-stats', (data) => {
      this.getSearchStats(data);
    });

    const data = [
      {
        page: '/',
        stats: pages['/'] ? pages['/'] : 0,
      },
      {
        page: '/deals',
        stats: pages['/deals'] ? pages['/deals'] : 0,
      },
      {
        page: '/dashboard',
        stats: pages['/dashboard'] ? pages['/dashboard'] : 0,
      },
    ];

    return (
      <div className="dashboard-container">
        <div style={{ display: 'flex', justifyContent: 'center' }}>
          <Card
            hoverable
            style={{ width: 240, marginRight: '1rem' }}
            cover={(
              <div className="eye-icon-container">
                <Icon type="eye" style={{ fontSize: '10rem', color: '#fff' }} />
              </div>
            )}
          >
            <Meta title="Site Visits" description={`${visits}`} />
          </Card>

          <Card
            hoverable
            style={{ width: 240, marginRight: '1rem' }}
            cover={(
              <div className="user-icon-container">
                <Icon type="user" style={{ fontSize: '10rem', color: '#fff' }} />
              </div>
            )}
          >
            <Meta title="Active Users" description={`${activeUsers}`} />
          </Card>

          <Card
            hoverable
            style={{ width: 240, marginRight: '1rem' }}
            cover={(
              <div className="search-icon-container">
                <Icon type="search" style={{ fontSize: '10rem', color: '#fff' }} />
              </div>
            )}
          >
            <Meta title="Searches" description={`${searches}`} />
          </Card>
        </div>

        <h2 style={{ color: 'gray' }}>Page stats</h2>

        <List
          grid={{
            gutter: 16,
            xs: 1,
            sm: 2,
            md: 4,
            lg: 4,
            xl: 6,
            xxl: 3,
          }}
          dataSource={data}
          renderItem={item => (
            <List.Item>
              <Card title={item.page}>{`${item.stats}`}</Card>
            </List.Item>
          )}
        />
      </div>
    );
  }
}

export default DashboardPage;
