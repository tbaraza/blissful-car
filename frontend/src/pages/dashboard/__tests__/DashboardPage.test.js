import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import DashboardPage from '../DashboardPage';

Enzyme.configure({ adapter: new Adapter() });
let wrapper;
describe('<DashboardPage/>', () => {
  it('renders correctly', () => {
    wrapper = shallow(<DashboardPage />);
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
