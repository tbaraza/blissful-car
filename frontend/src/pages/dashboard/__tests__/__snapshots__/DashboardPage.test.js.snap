// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardPage/> renders correctly 1`] = `
<div
  className="dashboard-container"
>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <div
      className="ant-card ant-card-bordered ant-card-hoverable"
      style={
        Object {
          "marginRight": "1rem",
          "width": 240,
        }
      }
    >
      <div
        className="ant-card-cover"
      >
        <div
          className="eye-icon-container"
        >
          <i
            className="anticon anticon-eye"
            style={
              Object {
                "color": "#fff",
                "fontSize": "10rem",
              }
            }
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="eye"
              fill="currentColor"
              height="1em"
              onClick={undefined}
              style={undefined}
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 0 0 0 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
              />
            </svg>
          </i>
        </div>
      </div>
      <div
        className="ant-card-body"
        style={Object {}}
      >
        <div
          className="ant-card-meta"
        >
          <div
            className="ant-card-meta-detail"
          >
            <div
              className="ant-card-meta-title"
            >
              Site Visits
            </div>
            <div
              className="ant-card-meta-description"
            >
              0
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-card ant-card-bordered ant-card-hoverable"
      style={
        Object {
          "marginRight": "1rem",
          "width": 240,
        }
      }
    >
      <div
        className="ant-card-cover"
      >
        <div
          className="user-icon-container"
        >
          <i
            className="anticon anticon-user"
            style={
              Object {
                "color": "#fff",
                "fontSize": "10rem",
              }
            }
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="user"
              fill="currentColor"
              height="1em"
              onClick={undefined}
              style={undefined}
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M858.5 763.6a374 374 0 0 0-80.6-119.5 375.63 375.63 0 0 0-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 0 0-80.6 119.5A371.7 371.7 0 0 0 136 901.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 0 0 8-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
              />
            </svg>
          </i>
        </div>
      </div>
      <div
        className="ant-card-body"
        style={Object {}}
      >
        <div
          className="ant-card-meta"
        >
          <div
            className="ant-card-meta-detail"
          >
            <div
              className="ant-card-meta-title"
            >
              Active Users
            </div>
            <div
              className="ant-card-meta-description"
            >
              0
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-card ant-card-bordered ant-card-hoverable"
      style={
        Object {
          "marginRight": "1rem",
          "width": 240,
        }
      }
    >
      <div
        className="ant-card-cover"
      >
        <div
          className="search-icon-container"
        >
          <i
            className="anticon anticon-search"
            style={
              Object {
                "color": "#fff",
                "fontSize": "10rem",
              }
            }
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="search"
              fill="currentColor"
              height="1em"
              onClick={undefined}
              style={undefined}
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M608 112c-167.9 0-304 136.1-304 304 0 70.3 23.9 135 63.9 186.5L114.3 856.1a8.03 8.03 0 0 0 0 11.3l42.3 42.3c3.1 3.1 8.2 3.1 11.3 0l253.6-253.6C473 696.1 537.7 720 608 720c167.9 0 304-136.1 304-304S775.9 112 608 112zm161.2 465.2C726.2 620.3 668.9 644 608 644s-118.2-23.7-161.2-66.8C403.7 534.2 380 476.9 380 416s23.7-118.2 66.8-161.2c43-43.1 100.3-66.8 161.2-66.8s118.2 23.7 161.2 66.8c43.1 43 66.8 100.3 66.8 161.2s-23.7 118.2-66.8 161.2z"
              />
            </svg>
          </i>
        </div>
      </div>
      <div
        className="ant-card-body"
        style={Object {}}
      >
        <div
          className="ant-card-meta"
        >
          <div
            className="ant-card-meta-detail"
          >
            <div
              className="ant-card-meta-title"
            >
              Searches
            </div>
            <div
              className="ant-card-meta-description"
            >
              0
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <h2
    style={
      Object {
        "color": "gray",
      }
    }
  >
    Page stats
  </h2>
  <div
    className="ant-list ant-list-split ant-list-grid"
  >
    <div
      className="ant-spin-nested-loading"
      style={null}
    >
      <div
        className="ant-spin-container"
      >
        <div
          className="ant-row"
          style={
            Object {
              "marginLeft": -8,
              "marginRight": -8,
            }
          }
        >
          <div
            className="ant-col-xs-24 ant-col-sm-12 ant-col-md-6 ant-col-lg-6 ant-col-xl-4 ant-col-xxl-8"
          >
            <div
              className="ant-list-item"
              style={
                Object {
                  "paddingLeft": 8,
                  "paddingRight": 8,
                }
              }
            >
              <div
                className="ant-list-item-content ant-list-item-content-single"
              >
                <div
                  className="ant-card ant-card-bordered"
                >
                  <div
                    className="ant-card-head"
                    style={Object {}}
                  >
                    <div
                      className="ant-card-head-wrapper"
                    >
                      <div
                        className="ant-card-head-title"
                      >
                        /
                      </div>
                    </div>
                  </div>
                  <div
                    className="ant-card-body"
                    style={Object {}}
                  >
                    0
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="ant-col-xs-24 ant-col-sm-12 ant-col-md-6 ant-col-lg-6 ant-col-xl-4 ant-col-xxl-8"
          >
            <div
              className="ant-list-item"
              style={
                Object {
                  "paddingLeft": 8,
                  "paddingRight": 8,
                }
              }
            >
              <div
                className="ant-list-item-content ant-list-item-content-single"
              >
                <div
                  className="ant-card ant-card-bordered"
                >
                  <div
                    className="ant-card-head"
                    style={Object {}}
                  >
                    <div
                      className="ant-card-head-wrapper"
                    >
                      <div
                        className="ant-card-head-title"
                      >
                        /deals
                      </div>
                    </div>
                  </div>
                  <div
                    className="ant-card-body"
                    style={Object {}}
                  >
                    0
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="ant-col-xs-24 ant-col-sm-12 ant-col-md-6 ant-col-lg-6 ant-col-xl-4 ant-col-xxl-8"
          >
            <div
              className="ant-list-item"
              style={
                Object {
                  "paddingLeft": 8,
                  "paddingRight": 8,
                }
              }
            >
              <div
                className="ant-list-item-content ant-list-item-content-single"
              >
                <div
                  className="ant-card ant-card-bordered"
                >
                  <div
                    className="ant-card-head"
                    style={Object {}}
                  >
                    <div
                      className="ant-card-head-wrapper"
                    >
                      <div
                        className="ant-card-head-title"
                      >
                        /dashboard
                      </div>
                    </div>
                  </div>
                  <div
                    className="ant-card-body"
                    style={Object {}}
                  >
                    0
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
